### PR TITLE
Other/rails 5.2 comptability

### PR DESCRIPTION
**Pull request by islam-taha**
Changed string block argument into proc to be compatible with rails 5, and fix specs.

### CODE REVIEW

File: lib/stator/version.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Stator
   MAJOR       = 0
-  MINOR       = 2
-  PATCH       = 2
+  MINOR       = 3
+  PATCH       = 7
```
Comment (choubacha) [inline lib/stator/version.rb]: Why are we on patch 7? This change would normally go back to patch 0 if we bump the minor.

File: lib/stator/transition.rb [hunk 2]
```diff
@@ -63,19 +63,24 @@ def callbacks(kind)
...
-          ) && (
-            #{@froms.inspect}.include?(self._stator(#{@namespace.inspect}).integration(self).state_was(#{options[:use_previous].inspect})) ||
-            #{@froms.inspect}.include?(::Stator::Transition::ANY)
-          ) && (
-            self._stator(#{@namespace.inspect}).integration(self).state == #{@to.inspect} ||
-            #{@to.inspect} == ::Stator::Transition::ANY
-          )
-        }
+
+      _namespace = @namespace
```
Comment (choubacha) [inline lib/stator/transition.rb:69]: I like this use of `_` prefix! 👍

**Review by choubacha — COMMENTED** [review]
I'd like to see a few things corrected before we merge your code:

1. Let's find out what version of stator is actually running in v3. It kinda looks like we're on v0.1.0, which is a minor version behind. I'd like to get us running the most up-to-date code on production first.
2. We've got to rebase master and bring it up to date with upstream.
3. Propose this change against the upstream and ping the owner to review/accept/publish the change.
4. Upgrade the gem in v3.

At step 3, while we are waiting for the owner to accept the change, we can reference the gem by git branch for testing.

File: lib/stator/version.rb (hunk 1 above)
Comment (islam-taha) [reply lib/stator/version.rb]: My bad, i'll fix this

**Pull request closed without merging**